core/templates/pages/learner-dashboard-page/community-lessons-tab.component.spec.ts

```diff
@@ -779,4 +804,38 @@ describe('Community lessons tab Component', () => {
 
     expect(modalSpy).toHaveBeenCalled();
   }));
+
+  it('should return true when there are no lessons', () => {
+    expect(component.isLearnerStateEmpty()).toBeTrue();
+  });
+
+  it('should return false when there are in-progress lessons', () => {
+    component.totalIncompleteLessonsList = [
+      LearnerExplorationSummary.createFromBackendDict(sampleExploration),
+    ];
+    fixture.detectChanges();
+
+    expect(component.isLearnerStateEmpty()).toBeFalse();
```

Comment:
Should we add a pre-check verifying if the learner state is empty here? Ditto for the below unit tests.

Reply:
I thought totalCompletedLessonsList & totalIncompleteLessonsList were initialized as empty Arrays in beforeEach (for some reason they weren't previously). Added them so we can assume the dependent variables are empty. Done.